src/modules/load_mon/LoadMon.cpp

```diff
@@ -142,17 +142,75 @@ void LoadMon::cpuload()
 	const float interval_idletime = total_runtime - _last_idle_time;
 #endif
 
-	// get ram usage
-	struct mallinfo mem = mallinfo();
-	float ram_usage = (float)mem.uordblks / mem.arena;
-
 	cpuload_s cpuload{};
 #if defined(__PX4_LINUX)
+	/* following calculation is based on free(1)
+	 * https://gitlab.com/procps-ng/procps/-/blob/master/proc/sysinfo.c */
+	char line[256];
+	int32_t kb_main_total = -1, kb_main_free = -1, kb_page_cache = -1, kb_slab_reclaimable = -1, kb_main_buffers = -1;
```

Comment:
Nitpick, multiple declarations per line are often discouraged (MISRA, etc).

Reply:
Thanks for this kind of teaching. It' s my first time hearing MISRA and the convention.